webrender/src/render_backend.rs

```diff
@@ -77,6 +77,11 @@ struct Document {
     // the first frame would produce inconsistent rendering results, because
     // scroll events are not necessarily received in deterministic order.
     render_on_scroll: Option<bool>,
+    // A helper flag to prevent any hit-tests from happening between calls
+    // to build_scene and rendering the document. In between these two calls,
+    // hit-tests produce inconsistent results because the clip_scroll_tree
+    // is out of sync with the display list.
```

Comment:
this is a little bit worrying, could we get a little more details on what's going on there and why?

Reply:
Does the comment at https://github.com/servo/webrender/issues/2099#issuecomment-347161775 help? I'm not sure what kind of details you're looking for exactly.